sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -223,10 +255,24 @@ public void testReadDisplayData() {
   }
 
   @Test
+  public void testReadDisplayDataWithGqlQuery() {
+    DatastoreV1.Read read =  DatastoreIO.v1().read()
+        .withProjectId(PROJECT_ID)
+        .withGqlQuery(GQL_QUERY)
+        .withNamespace(NAMESPACE);
+
+    DisplayData displayData = DisplayData.from(read);
+
+    assertThat(displayData, hasDisplayItem("projectId", PROJECT_ID));
+    assertThat(displayData, hasDisplayItem("gqlQuery", GQL_QUERY));
+    assertThat(displayData, hasDisplayItem("namespace", NAMESPACE));
+  }
+
+  @Test
   @Category(RunnableOnService.class)
```

Comment:
I think `@NeedsRunner` is for tests that construct pipelines but don't run them (a runner is needed to construct a pipeline too), and `@RunnableOnService` is for tests that also run them. This one and a bunch of others here don't run the pipelines, so please change to `@NeedsRunner` as appropriate.